src/Components/Server/src/Circuits/RemoteJSRuntime.cs

```diff
@@ -22,9 +22,14 @@ internal class RemoteJSRuntime : JSRuntime
 
         public bool IsInitialized => _clientProxy is not null;
 
-        public RemoteJSRuntime(IOptions<CircuitOptions> options, ILogger<RemoteJSRuntime> logger)
+        public RemoteJSRuntime(
+            IOptions<CircuitOptions> circuitOptions,
+            IOptions<HubOptions> hubOptions,
+            ILogger<RemoteJSRuntime> logger)
         {
-            _options = options.Value;
+            _options = circuitOptions.Value;
+            SignalRMaxMessageSizeBytes = hubOptions.Value.MaximumReceiveMessageSize is null ?
+                SignalRMaxMessageSizeBytes : hubOptions.Value.MaximumReceiveMessageSize.Value;
```

Comment:
Apparently a `null` value is meant to mean "unlimited" ([source](https://github.com/dotnet/aspnetcore/issues/13466#issuecomment-525154822)), whereas this logic would set it to 32KB. That problem would be fixed if we changed the default in JSInterop to `long.MaxValue` as in my other comment.

Reply:
@BrennanConroy What's the scenario for someone removing the limit on MaximumReceiveMessageSize. Under which circumstances do we consider that to be ok and are willing to tell customers to do so.

Reply:
Someone might have an app that doesn't even accept requests from the outside world, e.g., some kind of airgapped kiosk that just uses Blazor Server as a UI (and I know for sure one of our customers does exactly that). If SignalR already allows unlimited incoming data, I don't see any reason not to honour that API.

Reply:
@SteveSandersonMS That's fine, I'm not arguing against it, we can always make any limit useless by giving it a high enough number, so its not problematic to have a way to remove it. 

I just want to get a better understanding of under which circumstances we make this recommendations because It's an improvement we can make to the Blazor Server docs.

Reply:
> Someone might have an app that doesn't even accept requests from the outside world, e.g., some kind of airgapped kiosk

Yeah, that's basically when we would "recommend" unlimited. When you completely trust the end user, or just don't care 😆